fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DownloadLanguagesPreferenceFragmentIntegration.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations.preferences.downloadlanguages
+
+import android.os.Build
+import androidx.annotation.VisibleForTesting
+import androidx.core.net.ConnectivityManagerCompat.RESTRICT_BACKGROUND_STATUS_ENABLED
+import androidx.core.net.ConnectivityManagerCompat.RESTRICT_BACKGROUND_STATUS_WHITELISTED
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.fenix.wifi.WifiConnectionMonitor
+
+/**
+ * Helper for observing WiFi connection and data saving mode.
+ */
+class DownloadLanguagesPreferenceFragmentIntegration(
+    private val wifiConnectionMonitor: WifiConnectionMonitor,
+    private val dataSaverAndWIfiListener: (Boolean) -> Unit,
```

Comment:
```suggestion
    private val dataSaverAndWifiChanged: (Boolean) -> Unit,
```
nit: There was typo here and just better naming since it's lambda function

Reply:
Done